src/app/modules/submission/encrypt-submission/encrypt-submission.middleware.ts

```diff
@@ -225,13 +229,26 @@ export const scanAndRetrieveAttachments = async (
   // have virus scanning enabled.
 
   // Step 3: Trigger lambda to scan attachments.
-  const triggerLambdaResult = await ResultAsync.combine(
+  const triggerLambdaResult: Result<
+    (
+      | false
+      | {
+          response: ParsedClearAttachmentResponse
+          cleanFile: ParseVirusScannerLambdaPayloadOkBody
+        }
+    )[],
+    VirusScanFailedError
+  > = await ResultAsync.combine(
     req.body.responses.map((response) => {
       if (isQuarantinedAttachmentResponse(response)) {
-        return triggerVirusScanning(response.answer)
+        return triggerVirusScanning(response.answer).map((result) => ({
+          response,
+          cleanFile: result.body,
+        }))
       }
+
       // If response is not an attachment, return ok.
```

Comment:
/nit: update comment here

Reply:
Sorry, which comment are you referring to? 😅